packages/syntax-core/src/RadioButton/RadioButton.tsx

```diff
@@ -53,20 +59,51 @@ const RadioButton = ({
   size?: "sm" | "md";
   /**
    * Value of the selected radio option
+   *
+   * @defaultValue the value of `label`
    */
   value?: string;
 }): ReactElement => {
   const [isFocused, setIsFocused] = useState(false);
+  const { isFocusVisible } = useFocusVisible();
+
+  const checkedCircle = (
+    <div
+      className={classnames(styles.background, styles[size], {
+        [styles.mdCheckedBorder]: size === "md",
+        [styles.smCheckedBorder]: size === "sm",
+        [styles.errorBorderColor]: error,
+        [styles.borderColor]: !error,
+        [styles.focusedRadioButton]: isFocused && isFocusVisible,
+      })}
+    />
+  );
+
+  const circle = (
+    <div
+      className={classnames(
+        styles.background,
+        styles[size],
+        styles.neutralBorder,
+        {
+          [styles.errorBorderColor]: error,
+          [styles.borderColor]: !error,
+          [styles.focusedRadioButton]: isFocused && isFocusVisible,
+        },
+      )}
+    />
+  );
```

Comment:
both circle and checkedCircle share some styles. Is there a way to express "base styles regardless of state" and "add to the `classnames` into the same div with additional styling depending on state"?

Reply:
More full fledged:

```jsx
  const sharedStyles = classnames(styles.background, styles[size], {
    [styles.errorBorderColor]: error,
    [styles.borderColor]: !error,
    [styles.focusedRadioButton]: isFocused && isFocusVisible,
  });
```

```
{checked ? (
        <div
          className={classnames(sharedStyles, {
            [styles.mdCheckedBorder]: size === "md",
            [styles.smCheckedBorder]: size === "sm",
          })}
        />
      ) : (
        <div className={classnames(sharedStyles, styles.neutralBorder)} />
      )}
```